### ISSUE TITLE
Typescript LSP server fails

### ISSUE THREAD

**Issue by ctron** [eclipse-che/che#9814]
### Description

Opening a typescript file creates the following error:

> Initializing Language Server for index.tx
> Timeout initializing error

### Reproduction Steps

Opening a ".ts" file in a typescript project.

**OS and version:**    

~~~
[root@centos-16gb-nbg1-1 ~]# uname -a
Linux centos-16gb-nbg1-1 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
[root@centos-16gb-nbg1-1 ~]# cat /etc/redhat-release 
CentOS Linux release 7.4.1708 (Core) 
[root@centos-16gb-nbg1-1 ~]# oc version
oc v3.9.0+ba7faec-1
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://XXX:8443
openshift v3.9.0+ba7faec-1
kubernetes v1.9.1+a0ce1bc657
~~~

**Diagnostics:** 

The following can be seen in the "dev-machine" log window:

~~~
2018/05/25 07:52:11 Start new terminal.
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:68)
	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:433)
Caused by: java.io.IOException: Stream closed
	at java.lang.Thread.run(Thread.java:748)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerManager.handle(RequestHandlerManager.java:212)
	at org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerManager$OneToOneHandler.handle(RequestHandlerManager.java:328)
	at org.eclipse.che.api.languageserver.registry.LanguageServerRegistryImpl.initialize(LanguageServerRegistryImpl.java:136)
	at com.sun.proxy.$Proxy89.initialize(Unknown Source)
java.lang.RuntimeException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:129)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:77)
	at org.eclipse.che.api.languageserver.registry.ServerInitializerImpl.initialize(ServerInitializerImpl.java:117)
	at org.eclipse.che.api.languageserver.service.LanguageServerInitializationHandler.lambda$new$0(LanguageServerInitializationHandler.java:37)
	at org.eclipse.che.api.core.jsonrpc.commons.RequestDispatcher.dispatch(RequestDispatcher.java:51)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.processRequest(JsonRpcMessageReceiver.java:97)
	at org.eclipse.che.api.core.jsonrpc.commons.reception.FunctionConfiguratorOneToOne.lambda$withFunction$0(FunctionConfiguratorOneToOne.java:75)
	at org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerManager$OneToOneHandler.access$1000(RequestHandlerManager.java:315)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.lambda$receive$0(JsonRpcMessageReceiver.java:72)
	at java.io.OutputStream.write(OutputStream.java:116)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:65)
	... 16 common frames omitted
	at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
2018-05-25 07:52:21,057[equestProcessor]  [ERROR] [oggingUncaughtExceptionHandler 31]   - Runtime exception caught in thread ServerSideRequestProcessor. Message: java.io.IOException: Stream closed
2018/05/25 07:53:01 Error occurs on sending ping message to websocket. websocket: close sent
2018/05/25 07:53:01 Failed to send websocket message: , due to occurred error websocket: close sent
~~~

~~~
~~~

**Comment by eivantsov** [comment]
@ctron installers do not work well on OpenShift, unless the installer script checks for a pre installed LS.

What stack did you use?

**Comment by ctron** [comment]
I actually used a factory: https://github.com/ctron/hono-example-demo-gauge/blob/master/.factory.json

**Comment by eivantsov** [comment]
@ctron so `eclipse/node`

I think we should add TS LS in that image and tune installer a little bit, so that it does not install anything (which is not always possible in OpenShift in runtime) if a particular npm is already installed

**Comment by ctron** [comment]
So it is a bug/missing feature after all?

**Comment by eivantsov** [comment]
It is difficult to say. The installers concept does not go well with OpenShift at all.

**Comment by ctron** [comment]
So what I (as a user) do to make it work?

**Comment by eivantsov** [comment]
@ctron try it again. It should work now

**Comment by ctron** [comment]
You mean using the master branch?

**Comment by eivantsov** [comment]
Just rerun the workspace

**Comment by ctron** [comment]
Awesome! It works now!

Coulnd't you fix #9800 the same way :wink:
